### ISSUE TITLE
Week1

### ISSUE THREAD

**Issue by sweaterr** [sweaterr/coursera-kaggle#1]
### Reading list
* https://www.datasciencecentral.com/profiles/blogs/random-forests-explained-intuitively
* http://arogozhnikov.github.io/2016/06/24/gradient_boosting_explained.html

**Comment by sweaterr** [comment]
Quiz
![image](https://user-images.githubusercontent.com/1518919/36632151-f90a642e-19c5-11e8-8fc7-64aba0aaee16.png)

![image](https://user-images.githubusercontent.com/1518919/36632153-fe7e8da4-19c5-11e8-9d95-c1edbf668ae6.png)

**Comment by sweaterr** [comment]
## Feature preprocessing and generation with respect to models

### scaling
* 트리 베이스 모델은 scaling에 영향을 받지 않지만, linear model, neural network 같은 것은 scaling에 영향을 받음
* 경사 강하 법은 적절한 스케일링 없이 미쳐 날뜀
* 어떤 면에서 최적화해야할 또다른 hyperparameter 임
* 종류
  * sklearn.preprocessing.MinMaxScaler
  * sklearn.preprocessing.StandardScaler
* outliers
  * 지나치게 다른 피쳐와 다른 example이 있을 때 linear model
<img src="https://user-images.githubusercontent.com/1518919/36767317-51697812-1c7d-11e8-9106-7ef43787cfb3.png" width="200">
  * cliping 해야 한다.
  *  rank 처리가 outlier 처리에 유용할 수 있다. `scipy.stats.rankdata`
<img src="https://user-images.githubusercontent.com/1518919/36767535-40a91a22-1c7e-11e8-9f48-342e641f6ab0.png" width="200">
* Log transform =>  `np.log(1+x)`
* Raising to the power < 1 =>  `np.sqrt(x + 2/3)`
* preprocessing을 다르게 한 피쳐들을 concat해서 쓸 때 성능이 좋을 수도 있다.

### Feature generation
* 기존에 있는 피쳐를 합성해서 새로운 피쳐로 만들기 
* Prior knowledge, EDA

**Comment by sweaterr** [comment]
## Categorical and ordinal features 
Label encoding 
* Alphabetical (sorted) => `sklearn.preprocessing.LabelEncoder` 
* Order of appearance => `Pandas.factorize`
* Frequency encoding
```python
encoding = titanic.groupby('Embarked').size()
encoding = encoding/len(titanic)
titanic['enc'] = titanic.Embarked.map(encoding)
```
* `from scipy.stats import rankdata`
* one-hot encoding, non-tree based models에 유용한다
* 클래스 합치기
<img src="https://user-images.githubusercontent.com/1518919/36768334-08db05d4-1c82-11e8-90e6-07908eaabff8.png" width="400">

**Comment by sweaterr** [comment]
## Datetime and coordinates

Datetime
* 주,달,계절,년,초,분,시간
* 다음 휴일까지 남은 날수 
* 날짜간 차이

Coordinates
* Interesting places
* centers of clusters
* aggregated statistics
